src/nvim/regexp_nfa.c

```diff
@@ -6064,10 +6070,17 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
       break;
 
     // Allow interrupting with CTRL-C.
-    fast_breakcheck();
+    line_breakcheck();
```

Comment:
@oni-link Thanks, `make unittest` triggers a segfault with the following backtrace:
```
#0  0x00007ffff6839ab4 in queue_remove (queue=0x663480) at /home/jurica/sources/neovim/src/nvim/event/queue.c:163
#1  0x00007ffff6839858 in queue_get (queue=0x663480) at /home/jurica/sources/neovim/src/nvim/event/queue.c:124
#2  0x00007ffff6839964 in queue_process_events (queue=0x663480) at /home/jurica/sources/neovim/src/nvim/event/queue.c:140
#3  0x00007ffff683c476 in loop_poll_events (loop=0x7ffff6afdc20 <loop>, ms=0) at /home/jurica/sources/neovim/src/nvim/event/loop.c:56
#4  0x00007ffff6822316 in os_breakcheck () at /home/jurica/sources/neovim/src/nvim/os/input.c:150
#5  0x00007ffff678aac1 in line_breakcheck () at /home/jurica/sources/neovim/src/nvim/misc1.c:2717
#6  0x00007ffff67635f7 in nfa_regmatch (prog=0x66fac0, start=0x66fbd8, submatch=0x7fffffffd660, m=0x7fffffffd850) at /home/jurica/sources/neovim/src/nvim/regexp_nfa.c:6073
#7  0x00007ffff6763792 in nfa_regtry (prog=0x66fac0, col=0, tm=0x0) at /home/jurica/sources/neovim/src/nvim/regexp_nfa.c:6145
#8  0x00007ffff6763f8f in nfa_regexec_both (line=0x663890 "path_test_file", startcol=0, tm=0x0) at /home/jurica/sources/neovim/src/nvim/regexp_nfa.c:6309
#9  0x00007ffff6764324 in nfa_regexec_nl (rmp=0x7fffffffdc50, line=0x663890 "path_test_file", col=0, line_lbr=false) at /home/jurica/sources/neovim/src/nvim/regexp_nfa.c:6461
#10 0x00007ffff67645f9 in vim_regexec_both (rmp=0x7fffffffdc50, line=0x663890 "path_test_file", col=0, nl=false) at /home/jurica/sources/neovim/src/nvim/regexp.c:7058
#11 0x00007ffff67647ca in vim_regexec (rmp=0x7fffffffdc50, line=0x663890 "path_test_file", col=0) at /home/jurica/sources/neovim/src/nvim/regexp.c:7097
#12 0x00007ffff680a446 in fname_match (rmp=0x7fffffffdc50, name=0x663890 "path_test_file", ignore_case=false) at /home/jurica/sources/neovim/src/nvim/buffer.c:1933
#13 0x00007ffff680a3a6 in buflist_match (rmp=0x7fffffffdc50, buf=0x676fe0, ignore_case=false) at /home/jurica/sources/neovim/src/nvim/buffer.c:1914
#14 0x00007ffff6809e7c in buflist_findpat (pattern=0x40467f08 "path", pattern_end=0x0, unlisted=0, diffmode=0, curtab_only=0) at /home/jurica/sources/neovim/src/nvim/buffer.c:1781
#15 0x000000000042078f in lj_vm_ffi_call ()
#16 0x000000000042ba88 in lj_ccall_func (L=L@entry=0x40000378, cd=<optimized out>) at lj_ccall.c:878
#17 0x000000000041c296 in lj_cf_ffi_meta___call (L=0x40000378) at lib_ffi.c:229
#18 0x000000000041e847 in lj_BC_FUNCC ()
#19 0x000000000040d4b6 in lua_pcall (L=L@entry=0x40000378, nargs=nargs@entry=7, nresults=-1, errfunc=errfunc@entry=2) at lj_api.c:1052
#20 0x0000000000403db0 in docall (L=0x40000378, narg=7, clear=0) at luajit.c:122
#21 0x0000000000404c76 in handle_script (n=<optimized out>, argv=<optimized out>, L=<optimized out>) at luajit.c:289
#22 pmain (L=0x40000378) at luajit.c:538
#23 0x000000000041e847 in lj_BC_FUNCC ()
#24 0x000000000040d598 in lua_cpcall (L=L@entry=0x40000378, func=func@entry=0x4043d0 <pmain>, ud=ud@entry=0x0) at lj_api.c:1074
#25 0x00000000004038d6 in main (argc=13, argv=0x7fffffffe268) at luajit.c:566

```